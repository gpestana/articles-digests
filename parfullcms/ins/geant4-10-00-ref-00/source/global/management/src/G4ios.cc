//
// ********************************************************************
// * License and Disclaimer                                           *
// *                                                                  *
// * The  Geant4 software  is  copyright of the Copyright Holders  of *
// * the Geant4 Collaboration.  It is provided  under  the terms  and *
// * conditions of the Geant4 Software License,  included in the file *
// * LICENSE and available at  http://cern.ch/geant4/license .  These *
// * include a list of copyright holders.                             *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.  Please see the license in the file  LICENSE  and URL above *
// * for the full disclaimer and the limitation of liability.         *
// *                                                                  *
// * This  code  implementation is the result of  the  scientific and *
// * technical work of the GEANT4 collaboration.                      *
// * By using,  copying,  modifying or  distributing the software (or *
// * any work based  on the software)  you  agree  to acknowledge its *
// * use  in  resulting  scientific  publications,  and indicate your *
// * acceptance of all terms of the Geant4 Software license.          *
// ********************************************************************
//
//
// $Id: G4ios.cc 70021 2013-05-22 07:55:29Z gcosmo $
//
// 
// --------------------------------------------------------------
//      GEANT 4 class implementation file
//
// G4ios.cc
//
// History 1998 Nov. 3 Masayasu Nagamatu

#include "G4ios.hh"
#include "G4strstreambuf.hh"

#ifdef G4MULTITHREADED
  G4ThreadLocal G4strstreambuf *G4coutbuf_p = 0;
  G4ThreadLocal G4strstreambuf *G4cerrbuf_p = 0;
  G4ThreadLocal std::ostream *G4cout_p = 0;
  G4ThreadLocal std::ostream *G4cerr_p = 0;
  #define G4coutbuf (*G4coutbuf_p)
  #define G4cerrbuf (*G4cerrbuf_p)
  #define G4cout (*G4cout_p)
  #define G4cerr (*G4cerr_p)

  void G4iosInitialization()
  {
    if (G4coutbuf_p == 0) G4coutbuf_p = new G4strstreambuf;
    if (G4cerrbuf_p == 0) G4cerrbuf_p = new G4strstreambuf;
    if (G4cout_p == 0) G4cout_p = new std::ostream(G4coutbuf_p);
    if (G4cerr_p == 0) G4cerr_p = new std::ostream(G4cerrbuf_p);
  }

  void G4iosFinalization()
  {
    delete G4cout_p;
    delete G4cerr_p; 
    delete G4coutbuf_p;
    delete G4cerrbuf_p;
  }

#else  // Sequential

  G4strstreambuf G4coutbuf;
  G4strstreambuf G4cerrbuf;
  std::ostream G4cout(&G4coutbuf);
  std::ostream G4cerr(&G4cerrbuf);

  void G4iosInitialization() {}
  void G4iosFinalization() {}

#endif
