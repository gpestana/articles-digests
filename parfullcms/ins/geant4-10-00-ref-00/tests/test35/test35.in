//---------------------------------------
//        Macro file for test30
//         examine PostStepDoIt
//        for hadronic physics
//      Created: 13.03.2007 A.Ivanchenko
//---------------------------------------
#verbose
0
#rad
#events
100
//--------Proton_processes
#particle
proton
//--------
#material
G4_C
#momentum(MeV/c)
3000
#pmax(MeV/c)
3500.
#nbins
70
#nmompi
12
#nanglepi
10
#anglespi
0.35 0.55 0.75 0.95 1.15 1.35 1.55 1.75 1.95 2.15
#mompi(MeV/c)
100 150 200 250 300 350 400 450 500 600 700 800
//--------
#generator
ftfp
//#verbose
//1
#generator
bertini
#exit
