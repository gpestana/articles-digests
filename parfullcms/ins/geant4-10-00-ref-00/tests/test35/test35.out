========================================================
======             HARP Test Start           ========
========================================================
### New verbose level 0
Number of events according to macro 100
###### Start new run # 0     #####
Material is selected: G4_C
Test30Physics entry
Process is created gen= ftfp
New generator and material= G4_C
Nucleus with N= 12  Z= 6
Mass from targetNucleus(MeV)= 11174.9
Mass from IonTable(MeV)=      11174.9
Secondary generator <ftfp> is initialized
The particle:  proton
Ekin = 2.205031e+00 GeV
### factor  = 9.403219e-02    cross(b)= 2.591799e-01
### 0-th event start 
  User=0.02s Real=0.02s Sys=0s
#################################################################
###### Cross section per bin for pi+

## Next momentum bin 0  -  100  MeV/c

  0  0  0  0  0  0  0  0  0  0## Next momentum bin 100  -  150  MeV/c

  0  0  0  0  259.18  0  0  0  0  259.18## Next momentum bin 150  -  200  MeV/c

  0  0  0  259.18  0  259.18  0  0  0  0## Next momentum bin 200  -  250  MeV/c

  0  518.36  0  518.36  259.18  259.18  518.36  0  0  259.18## Next momentum bin 250  -  300  MeV/c

  148.103  0  0  0  259.18  0  0  259.18  0  0## Next momentum bin 300  -  350  MeV/c

  148.103  0  259.18  259.18  0  777.54  0  0  0  0## Next momentum bin 350  -  400  MeV/c

  148.103  0  0  518.36  0  0  0  0  0  0## Next momentum bin 400  -  450  MeV/c

  296.206  518.36  259.18  259.18  0  0  0  0  0  0## Next momentum bin 450  -  500  MeV/c

  0  259.18  0  0  0  0  0  0  0  0## Next momentum bin 500  -  600  MeV/c

  0  518.36  0  129.59  0  0  0  0  0  0## Next momentum bin 600  -  700  MeV/c

  74.0514  259.18  0  0  0  0  0  0  0  0## Next momentum bin 700  -  800  MeV/c

  148.103  129.59  0  0  0  0  0  0  0  0

## ds/do(mb/strad) for pi+ with momentum cut: 100  -  800  MeV/c
  59.2411  155.508  25.918  103.672  38.877  64.795  25.918  12.959  0  25.918
## ds/dp(mb/GeV) for pi+ with theta cut 0.35  -  2.15 radian
  0  570.573  552.845  2296.13  606.178  1366.75  488.561  723.919  141.43  405.001  141.43  70.7151
## cross_tot(bn)= 0.399034

#################################################################
## ds/do(mb/strad) for pi0 with momentum cut: 100  -  800  MeV/c
  29.6206  38.877  90.713  90.713  38.877  38.877  25.918  0  12.959  12.959

## ds/dp(mb/GeV) for pi0 with theta cut 0.35  -  2.15 radian
  161.392  604.828  845.37  976.32  1340.23  0  322.783  0  834.614  610.163  192.855  0
## cross_tot(bn)= 0.342648

#################################################################
## ds/do(mb/strad) for protons with momentum cut: 100  -  800  MeV/c
  96.2668  207.344  142.549  207.344  168.467  220.303  311.016  168.467  155.508  116.631

## ds/dp(mb/GeV) for pi0 with theta cut 0.35  -  2.15 radian
  2491.78  5308.83  4962.25  3775.21  2798.9  2855.2  3926.08  2365.91  2835.72  1712.69  1107.33  543.987
## cross_tot(bn)= 2.02698

#################################################################
## ds/do(mb/strad) for pi- with momentum cut: 100  -  800  MeV/c
  44.4308  51.836  51.836  12.959  25.918  38.877  12.959  12.959  0  0

## ds/dp(mb/GeV) for pi- with theta cut 0.35  -  2.15 radian
  0  324.133  1017.06  872.654  534.986  308.565  196.778  0  196.778  70.7151  0  70.7151
## cross_tot(bn)= 0.186691

## Next momentum bin 0  -  100  MeV/c
  0  0  0  0  0  0  0  0  0  0
## Next momentum bin 100  -  150  MeV/c
  148.103  0  0  0  0  0  0  259.18  0  0
## Next momentum bin 150  -  200  MeV/c
  0  259.18  0  259.18  0  259.18  259.18  0  0  0
## Next momentum bin 200  -  250  MeV/c
  0  0  0  0  518.36  259.18  0  0  0  0
## Next momentum bin 250  -  300  MeV/c
  148.103  259.18  518.36  0  0  0  0  0  0  0
## Next momentum bin 300  -  350  MeV/c
  148.103  0  0  0  0  259.18  0  0  0  0
## Next momentum bin 350  -  400  MeV/c
  0  0  259.18  0  0  0  0  0  0  0
## Next momentum bin 400  -  450  MeV/c
  0  0  0  0  0  0  0  0  0  0
## Next momentum bin 450  -  500  MeV/c
  0  0  259.18  0  0  0  0  0  0  0
## Next momentum bin 500  -  600  MeV/c
  148.103  129.59  0  0  0  0  0  0  0  0
## Next momentum bin 600  -  700  MeV/c
  74.0514  0  0  0  0  0  0  0  0  0
## Next momentum bin 700  -  800  MeV/c
  0  129.59  0  0  0  0  0  0  0  0
###### Start new run # 1     #####
Material is selected: G4_C
Test30Physics entry
Process is created gen= bertini
New generator and material= G4_C
Nucleus with N= 12  Z= 6
Mass from targetNucleus(MeV)= 11174.9
Mass from IonTable(MeV)=      11174.9
Secondary generator <bertini> is initialized
The particle:  proton
Ekin = 2.20503 GeV
### factor  = 0.0940322    cross(b)= 0.25918
### 0-th event start 
  User=0.04s Real=0.03s Sys=0s
#################################################################
###### Cross section per bin for pi+

## Next momentum bin 0  -  100  MeV/c

  0  0  0  0  0  0  129.59  129.59  0  129.59## Next momentum bin 100  -  150  MeV/c

  0  259.18  0  259.18  0  0  0  518.36  0  259.18## Next momentum bin 150  -  200  MeV/c

  148.103  0  0  259.18  0  0  259.18  518.36  518.36  0## Next momentum bin 200  -  250  MeV/c

  148.103  0  0  0  0  0  259.18  0  0  0## Next momentum bin 250  -  300  MeV/c

  148.103  0  0  0  259.18  0  0  0  0  0## Next momentum bin 300  -  350  MeV/c

  0  0  0  0  0  0  0  0  0  0## Next momentum bin 350  -  400  MeV/c

  0  0  259.18  259.18  0  0  0  0  0  0## Next momentum bin 400  -  450  MeV/c

  0  0  0  0  0  0  0  0  0  0## Next momentum bin 450  -  500  MeV/c

  148.103  259.18  0  259.18  0  0  0  0  0  0## Next momentum bin 500  -  600  MeV/c

  148.103  129.59  0  259.18  0  0  0  0  0  0## Next momentum bin 600  -  700  MeV/c

  74.0514  129.59  129.59  0  0  0  0  0  0  0## Next momentum bin 700  -  800  MeV/c

  74.0514  0  129.59  0  0  0  0  0  0  0

## ds/do(mb/strad) for pi+ with momentum cut: 100  -  800  MeV/c
  59.2411  51.836  38.877  77.754  12.959  0  25.918  51.836  25.918  12.959
## ds/dp(mb/GeV) for pi+ with theta cut 0.35  -  2.15 radian
  467.722  1322.51  1840.45  322.783  282.045  0  441.059  0  385.711  314.996  169.104  98.3889
## cross_tot(bn)= 0.334749

#################################################################
## ds/do(mb/strad) for pi0 with momentum cut: 100  -  800  MeV/c
  29.6206  51.836  77.754  64.795  51.836  64.795  142.549  12.959  0  12.959

## ds/dp(mb/GeV) for pi0 with theta cut 0.35  -  2.15 radian
  323.458  519.561  2120.61  1250.39  1733.78  526.326  1330.91  0  282.86  389.633  417.445  211.737
## cross_tot(bn)= 0.522449

#################################################################
## ds/do(mb/strad) for protons with momentum cut: 100  -  800  MeV/c
  81.4565  168.467  246.221  285.098  220.303  233.262  207.344  142.549  90.713  38.877

## ds/dp(mb/GeV) for pi0 with theta cut 0.35  -  2.15 radian
  3491.48  8062.36  8583.83  4642.08  1099.08  2094.32  2986.19  526.326  694.276  1338.85  543.916  314.996
## cross_tot(bn)= 2.00335

#################################################################
## ds/do(mb/strad) for pi- with momentum cut: 100  -  800  MeV/c
  0  38.877  25.918  51.836  51.836  64.795  51.836  38.877  12.959  25.918

## ds/dp(mb/GeV) for pi- with theta cut 0.35  -  2.15 radian
  0  570.573  568.414  2515.57  1404.13  0  694.276  0  1313.32  122.14  154.282  0
## cross_tot(bn)= 0.380956

## Next momentum bin 0  -  100  MeV/c
  0  0  0  0  0  0  0  0  0  0
## Next momentum bin 100  -  150  MeV/c
  0  0  0  0  259.18  0  0  0  0  259.18
## Next momentum bin 150  -  200  MeV/c
  0  0  0  259.18  0  0  0  259.18  0  0
## Next momentum bin 200  -  250  MeV/c
  0  0  518.36  0  518.36  259.18  0  518.36  259.18  259.18
## Next momentum bin 250  -  300  MeV/c
  0  259.18  0  0  0  518.36  518.36  0  0  0
## Next momentum bin 300  -  350  MeV/c
  0  0  0  0  0  0  0  0  0  0
## Next momentum bin 350  -  400  MeV/c
  0  259.18  0  259.18  0  259.18  0  0  0  0
## Next momentum bin 400  -  450  MeV/c
  0  0  0  0  0  0  0  0  0  0
## Next momentum bin 450  -  500  MeV/c
  0  259.18  0  259.18  259.18  0  518.36  0  0  0
## Next momentum bin 500  -  600  MeV/c
  0  0  0  129.59  0  0  0  0  0  0
## Next momentum bin 600  -  700  MeV/c
  0  0  0  0  0  129.59  0  0  0  0
## Next momentum bin 700  -  800  MeV/c
  0  0  0  0  0  0  0  0  0  0
###### End of test #####
